src/actions/composite.js

```diff
@@ -3,18 +3,24 @@ import { fetchNibrs } from './nibrs'
 import { fetchSummaries } from '../actions/summary'
 import { fetchUcrParticipation } from '../actions/participation'
 import history, { createNewLocation } from '../util/history'
+import { getPlaceId } from '../util/location'
+
 import {
   shouldFetchUcr,
   shouldFetchSummaries,
   shouldFetchNibrs,
 } from '../util/participation'
 
 const fetchData = () => (dispatch, getState) => {
-  const { filters } = getState()
-
-  if (shouldFetchUcr(filters)) dispatch(fetchUcrParticipation(filters))
-  if (shouldFetchSummaries(filters)) dispatch(fetchSummaries(filters))
-  if (shouldFetchNibrs(filters)) dispatch(fetchNibrs(filters))
+  const { filters, region, states } = getState()
+  if (region.loaded && states.loaded) {
+    if (!filters.placeId) {
+      filters.placeId = getPlaceId(filters, region.region, states.states);
```

Comment:
Why do you have `placeId` being passed through the URLs? It looks to me like duplicate state management (for example `explorer/state/florida/robbery?placeId=FL&since=2006&until=2016` has both "florida" and "FL") which seems like a great place for bugs to hang out. I would remove the placeId from the query parameters and just rely on the other values pulled from the URL.